crates/amaru-ledger/src/rules/context.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::BTreeMap;
+
+use amaru_kernel::{Hasher, KeepRaw, MintedTransactionBody, TransactionInput, TransactionOutput};
+
+// A slice here is a subset of the ledger state a ta moment in time
+pub type UtxoSlice = BTreeMap<TransactionInput, TransactionOutput>;
+
+// The BlockValidationContext is a collection of slices needed to validate a block
+#[derive(Default, Debug)]
+pub struct BlockValidationContext {
+    pub utxo_slice: UtxoSlice,
+    // TODO: add more slices as needed
+}
+
+impl BlockValidationContext {
+    pub fn update(
+        &mut self,
+        transaction_body: &KeepRaw<'_, MintedTransactionBody<'_>>,
+        transaction_is_valid: bool,
+    ) {
```

Comment:
I am not sure about this API; I think we're going to need something more _granular_ as you need to update the context while validating the transaction (e.g. add an available credential after processing a 'Reg' certificate).

Also, for this to work, the implementation should likely come from elsewhere and not be implemented by the context itself. The context defines the API, but the caller is going to implement the context API.